app/lib/actions/katello/foreman/content_update.rb

```diff
@@ -17,15 +18,18 @@ def plan(environment, content_view)
         def finalize
           environment  = ::Katello::KTEnvironment.find(input[:environment_id])
           content_view = ::Katello::ContentView.find(input[:content_view_id])
-          ::Katello::Foreman.update_puppet_environment(content_view, environment)
+          repository = ::Katello::Repository.find(input[:repository_id]) if input[:repository_id]
 
-          content_view.version(environment).repos(environment).each do |repo|
-            if distribution = repo.bootable_distribution
+          if content_view.default?
+            distribution = repository.try(:bootable_distribution)
+            if distribution
```

Comment:
Could combine the two statements:

`if (distribution = repository.try(:bootable_distribution)`